notebooks/Distance.ipynb

```diff
@@ -0,0 +1,812 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize algorithm\n",
+    "bfs = nk.distance.BFS(G, source, True, False, target)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run\n",
+    "bfs.run()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The distance from source to target node\n",
+    "print(bfs.distance(target))\n",
+    "# The number of shortest paths between the source node\n",
+    "print(bfs.numberOfPaths(target))\n",
+    "# Returns a shortest path from source to target\n",
+    "print(bfs.getPath(target))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Bidirectional BFS"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Bidirectional BFS algorithm explores the graph from both the source and target nodes until the two explorations meet. This version of BFS is more efficient than BFS when the target node is known.\n",
+    "\n",
+    "The [BidirectionalBFS(G, source, target, storePred=True)](https://networkit.github.io/dev-docs/python_api/distance.html?highlight=bidirec#networkit.distance.BidirectionalBFS) constructor expects a graph, the source and target nodes as mandatory parameters. The algorithm will also store the predecessors and reconstruct a shortest path from the source and the target if `storePred` is true."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize algorithm\n",
+    "biBFS = nk.distance.BidirectionalBFS(G, source, target)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run\n",
+    "biBFS.run()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The distance from source to target node\n",
+    "print(biBFS.getHops())\n",
+    "print(biBFS.getPath())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dijkstra "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dijkstra's algorithm finds the shortest path from a source node a target node. It algorithm creates a tree of shortest paths from the source to all other nodes in the graph.\n",
```

Comment:
```suggestion
    "Dijkstra's algorithm finds the shortest path from a source node a target node. It algorithm creates a tree of shortest paths from the source to all other nodes in the graph.\n",
```
Same as BFS, but Dijkstra does it also for weighted graphs. The descriptions of BFS and Dijkstra should be similar, the difference is that BFS only works with unweighted graphs.